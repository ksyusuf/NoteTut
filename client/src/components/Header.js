import React from 'react';

const Header = ({ onAddNote }) => (
  <header className="bg-white shadow-md p-4 rounded-xl m-4 mt-10">
    <div className="flex justify-between items-center">
      <h1 className="text-3xl font-bold text-gray-800">Tüm Notlarım</h1>
      <button
        className="bg-blue-600 text-white px-5 py-2 rounded-full shadow-lg hover:bg-blue-700 transition duration-200"
        onClick={onAddNote}
      >
        + Yeni Not Ekle
      </button>
    </div>
  </header>
);

export default Header;
