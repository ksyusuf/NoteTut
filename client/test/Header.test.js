import { screen, render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../src/components/Header';

describe('Header Component Testingg', () => {
  test('buton tıklanabilir olmalı ve doğru yazıyı içermeli', () => {
    const mockAddNote = jest.fn();
    render(<Header onAddNote={mockAddNote} />);

    // Butonu bulmak için role özelliğini kullan
    const button = screen.getByRole('button');
    // butona tıklama işlemi
    fireEvent.click(button);
    // buton başarıyla çağırıldı mı?
    expect(mockAddNote).toHaveBeenCalledTimes(1);

    // Butonun içindeki yazıyı kontrol et
    expect(button.textContent).toBe('+ Yeni Not Ekle');
  });

  test('Başlık doğru şekilde render edilmeli', () => {
    render(<Header onAddNote={jest.fn()} />);
    const headingElement = screen.getByRole('heading', { level: 1 }); // h1 etiketini al
    expect(headingElement.textContent).toBe('Tüm Notlarım');
  });

  test('Buton hover durumunda stil değişikliği olmalı', () => {
    render(<Header onAddNote={jest.fn()} />);
    const button = screen.getByRole('button');
    expect(button.getAttribute('class')).toContain('hover:bg-blue-700');

  });
  
  test('Header snapshot testi', () => {
    // header içerisinde diğer testleri geçip burada takılırsa css değişikliği yapılmış demektir.
    const { container } = render(<Header onAddNote={jest.fn()} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("başlık kontrolü", () => {
    render(<Header />); // burada header props ile bir işim olmadığı için onu yazmıyorum.
    const element = screen.getByText("Tüm Notlarım");
    expect(element).toBeInTheDocument()
  });
});