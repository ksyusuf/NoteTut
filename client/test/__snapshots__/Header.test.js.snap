// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Testingg Header snapshot testi 1`] = `
<header
  class="bg-white shadow-md p-4 rounded-xl m-4 mt-10"
>
  <div
    class="flex justify-between items-center"
  >
    <h1
      class="text-3xl font-bold text-gray-800"
    >
      Tüm Notlarım
    </h1>
    <button
      class="bg-blue-600 text-white px-5 py-2 rounded-full shadow-lg hover:bg-blue-700 transition duration-200"
    >
      + Yeni Not Ekle
    </button>
  </div>
</header>
`;
